File: LeetCode/JavaScript/#383 Ransom Note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(r, m) {
  var len_r = r.length;
  var len_m = m.length;
  if(len_r>len_m){return false;}
  r = r.split('');
  r.map(function(ele){
    m = m.replace(ele,"");
  });
  return len_m==m.length+len_r?true:false;
};

File: LeetCode/Python/#383 Ransom Note.py
#
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        return all(ransomNote.count(i)<=magazine.count(i) for i in set(ransomNote))


class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        return all(ransomNote.count(i)<=magazine.count(i) for i in string.ascii_lowercase)


class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        s,i=sorted(ransomNote),0
        for ele in sorted(magazine):
            i += ele = s[i] and i<len(s)
        return i==len(s)